touchforms/backend/touchcare.py

```diff
@@ -12,26 +12,242 @@
 from org.javarosa.core.model.instance import ExternalDataInstance
+from org.commcare.api.persistence import SqlSandboxUtils
+from org.commcare.core.sandbox import SandboxUtils
+from org.commcare.modern.process import FormRecordProcessorHelper as FormRecordProcessor
+from org.commcare.modern.parse import ParseUtilsHelper as ParseUtils
 from org.kxml2.io import KXmlParser
-
+import persistence
+from java.io import File
 from util import to_vect, to_jdate, to_hashtable, to_input_stream, query_factory
 from xcp import TouchFormsUnauthorized, TouchcareInvalidXPath, TouchFormsNotFound, CaseNotFound
 
 logger = logging.getLogger('formplayer.touchcare')
 
 
+def get_restore_url(criteria=None):
+    query_url = '%s?%s' % (settings.RESTORE_URL, urllib.urlencode(criteria))
+    return query_url
+
+
+def force_ota_restore(domained_username, auth):
+    username = domained_username.split("@")[0]
+    domain = domained_username.split("@")[1]
+    CCInstances({"username": username, "domain": domain, "host": settings.URL_HOST},
+                auth, force_sync=True, uses_sqlite=True)
+    result = {'status': 'OK'}
+    return result
+
+
+class CCInstances(CommCareInstanceInitializer):
+    def __init__(self, sessionvars, auth, restore_xml=None,
+                 force_sync=False, form_context=None, uses_sqlite=False):
+        self.vars = sessionvars
+        self.auth = auth
+        self.uses_sqlite = uses_sqlite
+
+        if self.uses_sqlite:
+            self.username = sessionvars['username'] + '@' + sessionvars['domain']
+            self.sandbox = SqlSandboxUtils.getStaticStorage(self.username, settings.SQLITE_DBS_DIRECTORY)
+            self.host = settings.URL_HOST
+            self.domain = sessionvars['domain']
+            self.query_func = query_factory(self.host, self.domain, self.auth, 'raw')
+            self.query_url = get_restore_url({'as': self.username, 'version': '2.0'})
+            CommCareInstanceInitializer.__init__(self, self.sandbox)
+
+            if force_sync or self.needs_sync():
+                self.perform_ota_restore(restore_xml)
+        else:
+            self.fixtures = {}
+            self.form_context = form_context or {}
+
+    def clear_tables(self):
+        db_name = self.username + ".db"
+        if os.path.isfile(db_name):
+            os.remove(db_name)
+        self.sandbox = SqlSandboxUtils.getStaticStorage(self.username, settings.SQLITE_DBS_DIRECTORY)
+        persistence.postgres_drop_sqlite(self.username)
+
+    def perform_ota_restore(self, restore=None):
+        self.clear_tables()
+        if not restore:
+            restore_xml = self.get_restore_xml()
+            ParseUtils.parseXMLIntoSandbox(restore_xml, self.sandbox)
+        else:
+            restore_file = restore
+            ParseUtils.parseFileIntoSandbox(File(restore_file), self.sandbox)
+        persistence.postgres_set_sqlite(self.username, 1)
+
+    def get_restore_xml(self):
+        payload = self.query_func(self.query_url)
```

Comment:
when i run the sms / ivr tests locally, they're failling because with sms / ivr surveys, it could be that a case is the "user" who is submitting the form, and in those cases this step here is failing